source/game/Game.gd

```diff
@@ -247,16 +247,26 @@ func _on_attack_selected(combatChoices: Dictionary, target: Unit) -> void:
 		if attacker_strikes > 0:
 			combatChoices['offense'].execute_each_turn(temp_current, target, true)
 			if randi() % 100 > target.get_defense():
+				combatChoices["offense"].execute_user_hit(temp_current, target, true)
```

Comment:
you could consider moving `offense` and `defense` strings to variables since they are reused multiple times